src/App.tsx

```diff
@@ -54,6 +56,8 @@ function App() {
         {path:"users-list",element:<UsersList/>},
```

Comment:
I suggest renaming the paths to 
```
 {path:"**users**",element:<UsersList/>},
 {path:"**projects**",element:<ProjectList/>},
 {path:"**projects/project-form**/:action",element:<ProjectForm/>},
{path:"**tasks**",element:<TasksList/>} 
{path:"**tasks**/tasks-form/:action",element:<TasksForm/>}
```

So what I did is removing list from the paths and making it plurals instead. I also add the correct hierarchy  to the pages, for example we reach the project-form from the projects page, so we need to make it `projects/project-form` having it as `project-form/:action` indicates that this is a main page in our application not a descendant one.